docs/confluence/advanced/grounding.md

```diff
@@ -0,0 +1,154 @@
+# Grounding 
+
+## Flag
+
+Grounding is on by default. To turn it off, use `--settings -smt_groundQuantifiers=false`
+
+ 
+## Grounding General Idea
+
+Grounding should make using quantifiers much easier on the solvers. For example,
+
+```
+assume forall x. f(x) = 0
+```
+
+A very natural init state axiom. Instead of giving it as is to the SMT-solvers, we automatically collect all instances of `f` in the formula we want to verify, e.g., `f(2), f(9), f(y + 3), f(z)`, and replace the original assert with:
+
+```
+assume f(2) = 0
+assume f(9) = 0
+assume f(y + 3) = 0
+assume f(z) = 0
+```
+
+Simple enough! You can of course write more complex quantified statements, and they will be grounded in a similar fashion. However, this approach has its limitations. It will never say a rule is valid when it is not (i.e., grounding is sound), but it may generate false counter-examples. Read below on how to minimize the chance of this happening. 
+
+Note that this works not only for assumes, but also for asserts, axioms, invariants, and anywhere else quantifiers are used. You can also use `exists` quantifiers.
+
+I’ve seen examples where this changes a timeout to 3 seconds, cases where it didn’t really change the running time, and a rare case where it performed much worse than standard quantification. but in general, this should work much better in terms of running time. 
```

Comment:
There are examples where this changes a timeout to 3 seconds, cases where it didn’t really change the running time, and a rare case where it performed much worse than standard quantification. However, in general, this should work much better in terms of running time.